src/sv-pipeline/scripts/SplitVariants.py

```diff
@@ -0,0 +1,72 @@
+#!/bin/python
+import argparse
+
+
+def process_bed_file(input_bed, N, bca=True):
+    condition_prefixes = {
+        'gt5kb': {
+            'condition': lambda line: (line[4] == 'DEL' or line[4] == 'DUP') and (int(line[2]) - int(line[1]) >= 5000)},
+        'lt5kb': {
+            'condition': lambda line: (line[4] == 'DEL' or line[4] == 'DUP') and (int(line[2]) - int(line[1]) < 5000)},
+        'bca': {'condition': lambda line: bca and (line[4] != 'DEL' and line[4] != 'DUP' and line[4] != 'INS')},
+        'ins': {'condition': lambda line: bca and line[4] == 'INS'}
+    }
+
+    current_lines = {prefix: [] for prefix in condition_prefixes.keys()}
+    current_counts = {prefix: 0 for prefix in condition_prefixes.keys()}
+    current_suffixes = {prefix: 'a' for prefix in condition_prefixes.keys()}
+
+    with open(input_bed, 'r') as infile:
+        for line in infile:
+            line = line.strip().split('\t')
+
+            for prefix, conditions in condition_prefixes.items():
```

Comment:
Each line will only match at most one condition, so you could `continue` after one condition is matched and the line is processed to prevent checking the others. Doesn't really matter though since the checks will be very quick